pkg/auth/providers/goauth/goauth_client.go

```diff
@@ -0,0 +1,62 @@
+package goauth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+)
+
+//GClient implements a httpclient for google oauth
+type GClient struct {
+	httpClient *http.Client
+}
+
+func (g *GClient) getUser(accessToken string, config *v3.GoogleOauthConfig) (Account, error) {
+	// userinfo endpoint doesn't require viewType param, non-admins and admins both can query this endpoint
+	respBytes, _, err := g.getFromGoogle(accessToken, config.UserInfoEndpoint)
+	if err != nil {
+		return Account{}, err
+	}
+
+	var goauthAccount Account
+	if err = json.Unmarshal(respBytes, &goauthAccount); err != nil {
+		return Account{}, err
+	}
+	return goauthAccount, nil
+}
+
+func (g *GClient) getFromGoogle(accessToken string, url string) ([]byte, int, error) {
+	var statusCode int
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, statusCode, err
+	}
+	req.Header.Add("Authorization", "Bearer "+accessToken)
```

Comment:
if it's the same set of headers used for github, we should put it in the common place

Reply:
The access token header is different for github and google, removed the user-agent header because calls work without it. So no common headers anymore